lib/src/sql/part.rs

```diff
@@ -114,11 +110,12 @@ impl fmt::Display for Part {
 			Part::All => f.write_str("[*]"),
 			Part::Last => f.write_str("[$]"),
 			Part::First => f.write_str("[0]"),
+			Part::Start(v) => write!(f, "{v}"),
 			Part::Field(v) => write!(f, ".{v}"),
 			Part::Index(v) => write!(f, "[{v}]"),
 			Part::Where(v) => write!(f, "[WHERE {v}]"),
 			Part::Graph(v) => write!(f, "{v}"),
-			Part::Value(v) => write!(f, "{v}"),
+			Part::Value(v) => write!(f, "[{v}]"),
```

Comment:
Is there a good reason to keep both `Part::Index` and `Part::Value`? Couldn't `Value::Number` act as an index?

Reply:
Yeah this is a good question. The only reason that I have kept them separate is because the `Value.cut()`, `Value,put()`, `Value.pick()` functions which are not asynchronous, and do not have access to the context or transaction, and therefore don't compute any values beforehand. Therefore they expect that any value is properly computed. These functions (which are used for more internal mutation, expect the value to be properly computed. So a `Part::Index` will always be a number, and a `Part::Field` will always be a field. However a `Part::Value` can be a string, parameter that needs to be computed, or a field name that needs to be computed.

I'll see if I can simplify this in a separate PR!